libs/sql-schema-describer/src/mssql.rs

```diff
@@ -0,0 +1,570 @@
+use super::*;
+use once_cell::sync::Lazy;
+use quaint::prelude::Queryable;
+use regex::Regex;
+use std::{
+    collections::{BTreeMap, HashMap, HashSet},
+    convert::TryInto,
+    sync::Arc,
+};
+
+static DEFAULT_INT: Lazy<Regex> = Lazy::new(|| Regex::new(r"\(\((.*)\)\)").unwrap());
+static DEFAULT_NON_INT: Lazy<Regex> = Lazy::new(|| Regex::new(r"\((.*)\)").unwrap());
+
+pub struct SqlSchemaDescriber {
+    conn: Arc<dyn Queryable + Send + Sync + 'static>,
```

Comment:
off-topic side note: we are re-arcing a `Quaint` struct that is already an `Arc`, I'd like to change that at some point (so it takes a `Quaint` instead.

Reply:
That would be great. I tried not to refactor here even when I wanted so bad to do that, due to me knowing the refactoring of all this is coming quite soon.